src/soroban.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Soroban helper class
+ * formatting, parsing, and etc
+ * @class Soroban
+ */
+export class Soroban {
+  /**
+   * All arithmetic inside the contract is performed on integers to
+   * avoid potential precision and consistency issues of floating-point
+   *
+   * This function takes the smart contract value and its decimals (if the token has any) and returns a display value
+   * @param {bigint | number | string} amount - the token amount you want to display
+   * @param {number} decimals - specify how many decimal places a token has
+   * @returns {string} - display value
+   */
+  static formatTokenAmount(amount, decimals) {
+    let formatted = amount.toString();
+
+    if (decimals > 0) {
+      formatted = (amount / 10 ** decimals).toFixed(decimals).toString();
+
+      // Trim trailing zeros
+      while (formatted[formatted.length - 1] === '0') {
+        formatted = formatted.replace(/0+$/, '');
+      }
+
+      if (formatted.endsWith('.')) {
+        formatted = formatted.replace(/0+$/, '');
+      }
```

Comment:
You don't need the loop, and I think the second clause doesn't cut off the period? I **think** this whole thing can be a simple one-liner:

```js
formatted = formatted.replace(/\.?0+$/, '');
```